### PR TITLE
Don't generate compose for EUS images

### PR DESCRIPTION
**Pull request by apaplaus**
Freshmaker will not generate ODCS compose for unpublished images that
suppose to be rebuilt.

JIRA: CWFHEALTH-208

Signed-off-by: Andrei Paplauski <apaplaus@redhat.com>

### CODE REVIEW

File: freshmaker/handlers/koji/rebuild_images_on_rpm_advisory_change.py [hunk 1]
```diff
@@ -295,7 +296,8 @@ def _record_batches(self, batches, event, builds=None):
 
                 db.session.commit()
 
-                if state != ArtifactBuildState.FAILED.value:
+                is_unpublished_exception = is_in_unpublished_exceptions(image)
+                if state != ArtifactBuildState.FAILED.value and not is_unpublished_exception:
```
Comment (mprahl) [inline freshmaker/handlers/koji/rebuild_images_on_rpm_advisory_change.py:300]: To keep the same behavior, could you change the if statement below instead to something like this?
```python
if image["generate_pulp_repos"] or not (image["published"] or is_in_unpublished_exceptions(image)):
```

File: freshmaker/utils.py [hunk 2]
```diff
@@ -217,3 +217,20 @@ def get_ocp_release_date(ocp_version):
...
+
+    :param ContainerImage image: image that should be checked
+    :return: True if image is published in 'exceptions' repository, False otherwise
+    """
+    for exception in conf.unpublished_exceptions:
+        exc_repo = exception["repository"]
+        exc_reg = exception["registry"]
+        for image_repo in image["repositories"]:
+            if image_repo["repository"] == exc_repo and image_repo["registry"] == exc_reg:
+                return True
```
Comment (hluk) [inline freshmaker/utils.py:235]: A bit more easier to read IMO:

```python
    repo_reg_pairs = {
        (image_repo["repository"], image_repo["registry"])
        for image_repo in image["repositories"]
    }
    return any(
        (exception["repository"], exception["registry"]) in repo_reg_pairs
        for exception in conf.unpublished_exceptions
    )
```

File: tests/test_utils.py [hunk 3]
```diff
@@ -40,6 +40,18 @@ def test_get_rebuilt_nvr(mock_time, rebuilt_nvr_release_suffix):
...
+@patch("freshmaker.config.Config.unpublished_exceptions",
+       new_callable=PropertyMock,
+       return_value=[{"repository": "some-repo", "registry": "some-registry"}])
+def test_is_in_unpublished_exceptions(exception):
+    images = [{"repositories": [{"registry": "some-registry", "repository": "repo"}]},
+              {"repositories": [{"registry": "registry", "repository": "some-repo"}]},
+              {"repositories": [{"registry": "some-registry", "repository": "some-repo"}]}]
+
+    results = list(map(is_in_unpublished_exceptions, images))
+    assert results == [False, False, True]
```
Comment (hluk) [inline tests/test_utils.py:52]:
```python
    assert is_in_unpublished_exceptions({"repositories": [{"registry": "some-registry", "repository": "repo"}]}) is False
    assert is_in_unpublished_exceptions({"repositories": [{"registry": "registry", "repository": "some-repo"}]}) is False
    assert is_in_unpublished_exceptions({"repositories": [{"registry": "some-registry", "repository": "some-repo"}]}) is True
```

But it's a good candidate to use `@pytest.mark.parametrize`.

Comment (apaplaus) [reply tests/test_utils.py:52]: When you use `@pytest.mark.parametrize` it generates a new test for every parameter, so with 3 parameters you will get 3 tests executed and addressed in the output. I prefer to have one output for one tested functionality so I would not use this fixture here. And secon

**Comment by mprahl** [comment]
Based on #61, I believe this will not be necessary. Let's keep this open until that one is merged.

**Comment by apaplaus** [comment]
Sure

**Comment by mprahl** [comment]
Closing this in favor of #61

**Pull request closed without merging**